### PR TITLE
Implement GetShaderInfoLog, GetProgramInfoLog and ValidateProgram

### PR DESCRIPTION
**Pull request by MortimerGoro**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/398)

### CODE REVIEW

File: webrender_traits/src/webgl.rs [hunk 1]
```diff
@@ -307,6 +314,8 @@ impl WebGLCommand {
                 gl::uniform_matrix_4fv(uniform_id, transpose, &v),
             WebGLCommand::UseProgram(program_id) =>
                 gl::use_program(program_id.get()),
+            WebGLCommand::ValidateProgram(program_id) =>
+                gl::validate_program(program_id.get()),
```
Comment (KiChjang) [inline webrender_traits/src/webgl.rs]: This doesn't have a return value?

Comment (MortimerGoro) [inline webrender_traits/src/webgl.rs]: No. Yo have to check the result using a separate function. I already tried in a JavaScript demo:

gl.getProgramParameter( program, gl.VALIDATE_STATUS)

**Comment by KiChjang** [comment]
@emilio or @pcwalton or @nox, is this the right way of adding additional traits to WR? I think we've already merged this into the main WR repo?

**Comment by jdm** [comment]
@KiChjang This is the WR repo...

**Comment by KiChjang** [comment]
Ugh, I keep on making stupids.

@bors-servo r+

Thanks!

**Pull request merged**